internal/operator-controller/rukpak/bundle/config.go

```diff
@@ -6,119 +30,317 @@ import (
 	"fmt"
 	"strings"
 
-	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/apimachinery/pkg/util/validation"
+	"github.com/santhosh-tekuri/jsonschema/v6"
 	"sigs.k8s.io/yaml"
+)
+
+const (
+	// configSchemaID is a name we use to identify the config schema when compiling it.
+	// Think of it like a file name - it just needs to be consistent.
+	configSchemaID = "config-schema.json"
 
-	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	// OwnNamespace mode: watchNamespace must equal install namespace
+	formatOwnNamespaceInstallMode = "ownNamespaceInstallMode"
+	// SingleNamespace mode: watchNamespace must differ from install namespace
+	formatSingleNamespaceInstallMode = "singleNamespaceInstallMode"
 )
 
+// ConfigSchemaProvider lets each bundle type describe what configuration it accepts.
+//
+// Different bundle types provide schemas in different ways:
+//   - registry+v1: builds schema from the operator's install modes
+//   - Helm: reads schema from values.schema.json in the chart
+//   - registry+v2: (future) will have its own way
+//
+// Note: We don't store this in the Config struct. We only need it when validating
+// the user's input. After that, the Config has the validated data and doesn't need
+// the schema anymore.
+type ConfigSchemaProvider interface {
+	// Get returns a JSON Schema describing what configuration is valid.
+	// Returns nil if this bundle type doesn't need configuration validation.
+	Get() (map[string]any, error)
+}
+
+// Config holds validated configuration data from a bundle.
+//
+// Different bundle types have different configuration options, so we store
+// the data in a flexible format and provide accessor methods to get values out.
+//
+// Why there's no Validate() method:
+// We validate configuration when creating a Config. If you have a Config object,
+// it's already been validated - you don't need to check it again. The unexported
+// 'data' field means you can't create a Config yourself; you have to use
+// UnmarshalConfig, which does the validation.
 type Config struct {
-	WatchNamespace *string `json:"watchNamespace"`
+	data map[string]any
```

Comment:
@perdasilva we are removing to make more Generic 
So, I think is important we move with it , if we want change it, before we promote Config API ( consequentely single/own ns )

Reply:
if Config has just single field, why do not declare it as

```golang
type Config ma[string]any
```